articlequality/utilities/fetch_text.py

```diff
@@ -68,27 +68,35 @@ def fetch_text(session, labelings, verbose=False):
             Print dots and stuff
 
     :Returns:
-        An `iterator` of labelings augmented with 'page_id', 'rev_id' and
-        'text'.  Note that labelings of articles that aren't found will not be
-        included.
+        An `iterator` of labelings augmented with 'page_id', 'rev_id', 'text',
+        'page_title' and 'talk_page_title'. Note that labelings of articles
+        that aren't found will not be included.
     """
 
     for labeling in labelings:
-        rev_doc = get_last_rev_before(session, labeling['page_title'],
-                                      labeling['timestamp'])
+        talk_info = get_talk_page_info(session, labeling['talk_page_id'])
+        try:
+            rev_doc = get_last_rev_before(session, talk_info['subjectid'],
+                                          labeling['timestamp'])
+        except (KeyError, TypeError):
+            # No subject page found
+            continue
 
         if rev_doc is None:
             if verbose:
                 sys.stderr.write("?")
                 sys.stderr.write(
-                    labeling['page_title'] + " " + labeling['timestamp'])
+                    labeling['talk_page_id'] + " " + labeling['timestamp'])
```

Comment:
This causes errors due to `labeling['talk_page_id']` not being a string. See https://github.com/wikimedia/articlequality/pull/131 for the fix.